--- CMakeLists.txt
+++ CMakeLists.txt
@@ -40,9 +40,9 @@
 # Dependencies
 # --------------
 # Eigen
-find_package(Eigen3 REQUIRED)
-include_directories (SYSTEM "${EIGEN3_INCLUDE_DIR}")
-set(Eigen_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+find_package(Eigen3 CONFIG REQUIRED NO_MODULE)
+get_target_property(Eigen_INCLUDE_DIRS Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(SYSTEM "${Eigen_INCLUDE_DIRS}")
 # --------------
 
 
